NotificationService/NotificationService.BusinessLibrary/Business/v1/EmailHandlerManager.cs

```diff
@@ -249,5 +249,27 @@ public async Task<IList<NotificationResponse>> ResendEmailNotifications(string a
             this.logger.TraceInformation($"Finished {nameof(this.ResendEmailNotifications)} method of {nameof(EmailHandlerManager)}.");
             return notificationResponses;
         }
+
+        /// <inheritdoc/>
+        public async Task<IList<NotificationResponse>> ResendEmailNotificationsByDateRange(string applicationName, DateTimeRange dateRange)
+        {
+            this.logger.TraceInformation($"Started {nameof(this.ResendEmailNotificationsByDateRange)} method of {nameof(EmailHandlerManager)}.");
+            var maxAllowedDaysInRange = (double)this.configuration.GetValue(typeof(double), Constants.AllowedMaxResendDays);
+            if (dateRange != null && (dateRange.EndDate - dateRange.StartDate).TotalDays >= maxAllowedDaysInRange)
+            {
+                throw new DataException($"Date-range must not be less or equal to {maxAllowedDaysInRange}");
+            }
+
+            var failedNtificationEntities = await this.emailManager.GetEmailNotificationsByDateRange(applicationName, dateRange).ConfigureAwait(false);
+            if (failedNtificationEntities == null || failedNtificationEntities.Count == 0)
+            {
+                return null;
+            }
+
+            var notificationIds = failedNtificationEntities.Select(notificationEntity => notificationEntity.NotificationId);
+            var result = await this.ResendEmailNotifications(applicationName, notificationIds.ToArray()).ConfigureAwait(false);
```

Comment:
Here, this should ignore the sent items. If api call timeout, and user hits again, it should not send again. So, Make sure you IgnoreAlreadyent items to true

Reply:
changes done as suggested.